### DISCUSSION TITLE
Unable to save changes when not signed in

### DISCUSSION THREAD

**Discussion by sj-rai** [codesandbox/codesandbox-client#6537] [Q&A] [answered]
When I have not signed in, codesandbox only allows upto 3 sandboxes. But I see it's creating a fork of my sandbox every time I make a save, and hence after I save three times, every time I try to save my changes, I get the error 
`Failed to save 'fileName.jsx': Unknown Save Error`,
 along with
 `You need to sign in to create more than 3 sandboxes`.
 Is this expected behaviour? Is it supposed to create a fork for every time I save my file?

Thanks

**Comment by JamesACS** [comment, marked as answer]
Hey Sanjoth, 

You will need to log in to save your Sandboxes to your user. If you're not logged in, the client doesn't have anything to save the work too.
